### ISSUE TITLE
model based clustering

### ISSUE THREAD

**Issue by rdewijn** [tercen/MTvC_app#2]
@ginberg: the default setting "auto" for the number of clusters in the model based clustering app does not appear to work properly, at least not in the app example.

**Comment by ginberg** [comment]
hi @rdewijn, I have looked into this and for the app example we get a different number of clusters indeed. I found that the mean function in Bionavigator and Tercen give slightly different results (in the order of 1e-05) and this causes the principal components and eventually the clusters to be different. I heard from Faris that the mean in Bionavigator is implemented in Smalltalk, could that be the reason the mean result is different? Any idea how we should solve this?

**Comment by rdewijn** [comment]
I am surprised that a mean function gives different results up to 1e-5. Means there is mistake somewhere. Absolutely sure you are comparing with the same data?

I am also surprised that a deviation that should be relatively minor should  has such a large effect on the results ?!

**Comment by ginberg** [comment]
difference after row scale operator. This small diff seems to cause the difference in number of clusters
[row_scale_diff.csv](https://github.com/tercen/MTvC_app/files/7978617/row_scale_diff.csv)

**Comment by ginberg** [comment]
@rdewijn we just did test the clustering on Bionavigator using R 4.0.4 (same R version as we using on Tercen)

- mclust latest version (5.4.9), we are getting 4 clusters
- mclust 5.2 (version in BN) we are getting 7 clusters
Also, when I am downgrading the mclust version on Tercen to 5.2, I am getting 7 clusters. So the result of the left branch in this workflow has been explained.
https://stage.tercen.com/pamgene/w/9182edbb22c84278dd63d2517e4f7c14

Should we downgrade the mclust to 5.2?

**Comment by rdewijn** [comment]
@ginberg @fnaji , 
- Do these 7 clusters look similar to the clusters obtained with BN?
- I think the clustering with 5.4.9. looks weird.
Maybe a good idea to downgrade to 5.2, in partcicular when that gives good correspondence with Bionavigator.

**Comment by ginberg** [comment]
@rdewijn yes, these clusters look similar, see below. Only in the last three clusters there is an extra straight line, but I think this might be a [drawing issue](https://github.com/tercen/sci/issues/416)

Bionavigator:
![BN_clusters](https://user-images.githubusercontent.com/7089667/152748461-ff95b56d-2dbc-4b16-9321-c710b587c0af.png)

Tercen:
![tercen_clusters](https://user-images.githubusercontent.com/7089667/152748467-d10c822a-fa83-47f4-9cd0-709dffc70eed.png)

**Comment by rdewijn** [comment]
OK, can we make sure that these horizontal lines are because of a different issue?

**Comment by ginberg** [comment]
@rdewijn these horizontal lines were indeed because of a different issue, it is fixed now by Alex. See github https://github.com/tercen/sci/issues/416 and here the view without these lines
https://stage.tercen.com/pamgene/w/9182edbb22c84278dd63d2517e4f7c14/ds/692fcfe3-259c-417e-8417-c3543cbf89fe
